app/models/script.rb

```diff
@@ -0,0 +1,102 @@
+require 'securerandom'
+
+class Script < ActiveRecord::Base
+  belongs_to :project
+  has_many :jobs, dependent: :destroy
+  validates :frames, presence: true, format: { with: /\d+\-\d+/ }
+
+  attr_accessor :project_dir
+
+  class << self
+    def never_submitted_status
+      'not submitted'
+    end
+
+    def default_cluster
+      'owens'
+    end
+  end
+
+  def submit
+    cluster_ok?
```

Comment:
Instead of `cluster_ok?` to prevent submitting to Ruby or Pitzer, the way we have done this in other apps is to remove invalid clusters from the constant that stores the list of clusters. In ActiveJobs and Job Composer this is currently done in an initializer https://github.com/OSC/ood-activejobs/blob/1968f5b28c00a48672a99bbf145992af72f6058b/config/initializers/ood_appkit.rb#L4-L6. However, here it would be preferable to add it to the configuration singleton so it could be something like `Configuration.clusters` and then the method would look like

```ruby
def clusters
  @clusters ||=   OodCore::Clusters.new(OodAppkit.clusters.select { |c| c.job_allow? && c.id == "owens" })
end
```